index.js

```diff
@@ -1,45 +1,57 @@
 const identity = x => x;
 const getUndefined = () => {};
 const filter = () => true;
-function createRavenMiddleware(Raven, options = {}) {
-  // TODO: Validate options.
-  const {
-    breadcrumbDataFromAction = getUndefined,
-    actionTransformer = identity,
-    stateTransformer = identity,
-    breadcrumbCategory = "redux-action",
-    filterBreadcrumbActions = filter,
-    getUserContext
-  } = options;
+const createRavenMiddleware = (function() {
+  let called = false;
+  return (Raven, options = {}) => {
+    // TODO: Validate options.
+    const {
+      breadcrumbDataFromAction = getUndefined,
+      actionTransformer = identity,
+      stateTransformer = identity,
+      breadcrumbCategory = "redux-action",
+      filterBreadcrumbActions = filter,
+      getUserContext
+    } = options;
 
-  return store => {
-    let lastAction;
-
-    Raven.setDataCallback((data, original) => {
-      const state = store.getState();
-      data.extra.lastAction = actionTransformer(lastAction);
-      data.extra.state = stateTransformer(state);
-      if (getUserContext) {
-        data.user = getUserContext(state);
+    if (!called) called = true;
+    else {
+      if (Raven._globalOptions.dataCallback instanceof Function) {
+        console.warn(
+          "Possible memory leak detected! Refer to https://github.com/captbaritone/raven-for-redux."
```

Comment:
As I mentioned above, this is about more than just the memory leak. Maybe just something about calling it twice not being supported?